src/components/Select2.js

```diff
@@ -69,6 +69,10 @@ export default class Select2 extends Component {
       this.initSelect2(false);
     }
 
+    if (!shallowEqualFuzzy(prevProps.options, this.props.options)) {
+      this.el.select2(this.props.options);
```

Comment:
When updating options you should consider following:
https://github.com/rkit/react-select2-wrapper/blob